crates/external_services/src/grpc_client.rs

```diff
@@ -0,0 +1,238 @@
+use std::fmt::Debug;
+
+pub use common_utils::errors::CustomResult;
+use common_utils::{ext_traits::OptionExt, id_type, transformers::ForeignFrom};
+pub use dynamic_routing::{
+    success_rate_calculator_client::SuccessRateCalculatorClient, CalSuccessRateConfig,
+    CalSuccessRateRequest, CalSuccessRateResponse,
+    CurrentBlockThreshold as DynamicCurrentThreshold, LabelWithStatus,
+    UpdateSuccessRateWindowConfig, UpdateSuccessRateWindowRequest, UpdateSuccessRateWindowResponse,
+};
+use error_stack::ResultExt;
+use hyperswitch_interfaces::api::api_models::routing::{
+    CurrentBlockThreshold, DynamicRoutingConfig, DynamicRoutingConfigBody, RoutableConnectorChoice,
+    RoutableConnectorChoiceWithStatus,
+};
+use router_env::logger;
+use tonic::transport::Channel;
+#[allow(
+    missing_docs,
+    unused_qualifications,
+    clippy::unwrap_used,
+    clippy::as_conversions
+)]
+pub mod dynamic_routing {
+    tonic::include_proto!("success_rate");
+}
+use serde;
+
+pub type DRResult<T> = CustomResult<T, DRError>;
+
+// The trait Success Based Dynamic Routing would have all the functions required to support the calculation and updation window
+#[async_trait::async_trait]
+pub trait SuccessBasedDynamicRouting: dyn_clone::DynClone + Send + Sync {
```

Comment:
Can we have a proper module hierarchy here? move these DR related types and implementation under grpc_client/dynamic_routing module